Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -246,6 +249,55 @@ namespace AZ::DocumentPropertyEditor
             }
         }
 
+        void VisitValueWithSerializedPath(Reflection::IObjectAccess& access, const Reflection::IAttributes& attributes)
+        {
+            const AZ::TypeId valueType = access.GetType();
+            void* valuePointer = access.Get();
+
+            rapidjson::Document serializedValue;
+            auto result = JsonSerialization::Store(serializedValue, serializedValue.GetAllocator(), valuePointer, nullptr, valueType);
+
+            AZ::Dom::Value instancePointerValue;
+            auto outputWriter = instancePointerValue.GetWriteHandler();
+            auto convertToAzDomResult = AZ::Dom::Json::VisitRapidJsonValue(serializedValue, *outputWriter, AZ::Dom::Lifetime::Temporary);
+            VisitValue(
+                instancePointerValue,
+                access.Get(),
+                attributes,
+                [valuePointer, valueType, this](const Dom::Value& newValue)
+                {
+                    void* marshalledPointer = AZ::Dom::Utils::TryMarshalValueToPointer(newValue, valueType);
+                    rapidjson::Document serializedValue;
+                    auto result =
+                        JsonSerialization::Store(serializedValue, serializedValue.GetAllocator(), marshalledPointer, nullptr, valueType);
+
+                    JsonDeserializerSettings deserializeSettings;
+                    deserializeSettings.m_serializeContext = m_serializeContext;
+                    // now deserialize that value into the original location
+                    JsonSerialization::Load(valuePointer, valueType, serializedValue, deserializeSettings);
+
+                    AZ::Dom::Value newInstancePointerValue;
+                    auto outputWriter = newInstancePointerValue.GetWriteHandler();
+                    auto convertToAzDomResult =
+                        AZ::Dom::Json::VisitRapidJsonValue(serializedValue, *outputWriter, AZ::Dom::Lifetime::Temporary);
+                    return newInstancePointerValue;
+                },
+                false,
+                false);
+        }
+
+        bool IsInspectorOverrideManagementEnabled()
```

Comment:
I don't think the ReflectionAdapter should need to know about Inspector/override specific concepts. Can you encapsulate this in a prefab or Insepctor specific class?

Reply:
I think in general we should figure out a delineation between a prefab adapter and a general reflection adapter - the most obvious to my mind is making `PrefabAdapter` a subclass of `ReflectionAdapter` and exposing these touch-points via virtual methods that are a no-op for reflection, but I don't personally care too much as long as there's some kind of divide. (See @alexmontAmazon , sometimes I DO lean towards inheritance over composition!)

Reply:
This helper function simply checks whether a particular feature flag is enabled or not. It doesn't have any prefab specific logic in it. This is only temporary and will be removed once the work in https://github.com/o3de/sig-content/issues/112 is finished. 

I agree that it feels odd to have it in this class but it'd be odd to have it anywhere else in AzFramework as well since everything about prefabs is in AzToolsFramework. So it's better to keep it here locally than make it a common utils method in AzFramework elsewhere. See https://github.com/o3de/o3de/pull/14507#discussion_r1101914970 for related discussion on this.

Reply:
ok, but you have it copy-pasted in two places, and this one is a file called ReflectionAdapter that doesn't need to know anything about prefabs! Can you at least merge these two identical functions into one place and have them both call the same method?